packages/compose/src/hooks/use-focus-outside/index.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * External dependencies
+ */
+import type {
+	FocusEventHandler,
+	EventHandler,
+	MouseEventHandler,
+	TouchEventHandler,
+	FocusEvent,
+	MouseEvent,
+	TouchEvent,
+} from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Input types which are classified as button types, for use in considering
+ * whether element is a (focus-normalized) button.
+ */
+const INPUT_BUTTON_TYPES = [ 'button', 'submit' ];
+
+type FocusNormalizedButton =
+	| HTMLButtonElement
+	| HTMLLinkElement
+	| HTMLInputElement;
+
+/**
+ * Returns true if the given element is a button element subject to focus
+ * normalization, or false otherwise.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Clicking_and_focus
+ *
+ * @param  eventTarget The target from a mouse or touch event.
+ *
+ * @return Whether element is a button.
```

Comment:
Missing the return type? Also a language nit.

```suggestion
 * @return {boolean} Whether the element is a button.
```

Reply:
Wording improved in https://github.com/WordPress/gutenberg/pull/45317/commits/c9cade699d80454f64fc8ed26237601163df3bf8. For the return type, we can keep https://github.com/WordPress/gutenberg/pull/45317#discussion_r1007787050 open to discuss further in case.